birdhouse/pavics-compose.sh

```diff
@@ -85,30 +81,58 @@ for adir in $AUTODEPLOY_EXTRA_REPOS; do
 done
 export AUTODEPLOY_EXTRA_REPOS_AS_DOCKER_VOLUMES
 
-# we apply all the templates
-find $ALL_CONF_DIRS -name '*.template' |
-  while read FILE
-  do
-    DEST=${FILE%.template}
-    cat ${FILE} | envsubst "$VARS" | envsubst "$OPTIONAL_VARS" > ${DEST}
-  done
+BUILD_DIR="${BUILD_DIR:-"${COMPOSE_DIR}/build"}"
+
+COMPOSE_FILE="${BUILD_DIR}/docker-compose.yml"
+
+# Keep the compose project name as "birdhouse" by default (no matter where the build directory is located)
+export COMPOSE_PROJECT_NAME=${COMPOSE_PROJECT_NAME:-birdhouse}
 
 if [ x"$1" = x"up" ]; then
+  rm -r "${BUILD_DIR}"
   for adir in $ALL_CONF_DIRS; do
+    CONF_NAME="$(basename "${adir}")"
+    find "$adir" -type f |
+      while read -r FILE
+      do
+        RELATIVE_FILE_PATH=${FILE#${adir}}
+        DEST="${BUILD_DIR}/${CONF_NAME}/${RELATIVE_FILE_PATH#/}"
+        mkdir -p "$(dirname "${DEST}")"
+        ln "${FILE}" "${DEST}"
```

Comment:
@tlvu What if we try the hardlink with a fallback strategy to copy the files. That way, we don't lose the ability to host these on other filesystems

Reply:
Oh yes, that would be best.  Test for `ln` exit code, if it errors out, use regular `cp` instead.

Reply:
Could consider `rsync` as well instead of `cp` to skip copies where no deltas are detected.

Reply:
The build directory is emptied before this so there would never be a case where there are no deltas. If we didn't clear the build directory first, this could be useful. But I worry that we'd end up with a lot of leftover files between builds if the configuration changes.

Reply:
Could that be an option? Like allowing `CLEAN_BUILD_DIR=true pavics-compose up -d` to take advantage of quicker `rsync`? I think cleaning the build dir is valid to start from a fresh setup, but wiping it each time while developing the same feature is overkill.